### ISSUE TITLE
Update CMake version to at least 3.10

### ISSUE THREAD

**Issue by Parad0x84** [lz4/lz4#1550]
CMake says versions less than 3.10 are deprecated, so please update to a newer version of CMake

**Comment by lidavidm** [comment]
It seems this was incidentally fixed in https://github.com/lz4/lz4/commit/684ffb3f224a72528b8e50a7dce541317dce10ad but there's no release including this.

**Comment by Cyan4973** [comment]
Correct, this is fixed in `dev`. The fix will be part of the next release (no ship date set yet).
